Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js

```diff
@@ -0,0 +1,153 @@
+}
+
+export function destroy(element, elementId) {
+    const instances = _instances || {};
+    const instance = instances[elementId];
+
+    if (instance) {
+        if (instance.sigpad) {
+            instance.sigpad.off();
+        }
+
+        delete instances[elementId];
+    }
+}
+
+export function updateOptions(element, elementId, options) {
+    const instance = _instances[elementId];
+    if (instance && instance.sigpad && options) {
+        if (options.penColor.changed) {
+            instance.sigpad.penColor = options.penColor.value || "black";
+        }
+
+        if (options.minLineWidth.changed) {
+            instance.sigpad.minWidth = options.minLineWidth.value;
+        }
+
+        if (options.maxLineWidth.changed) {
+            instance.sigpad.maxWidth = options.maxLineWidth.value;
+        }
+
+        if (options.backgroundColor.changed) {
+            instance.sigpad.backgroundColor = options.backgroundColor.value || "rgba(0,0,0,0)";
+            const data = instance.sigpad.toData();
+            instance.sigpad.clear();
+            instance.sigpad.fromData(data);
+        }
+
+        if (options.velocityFilterWeight.changed) {
+            instance.sigpad.velocityFilterWeight = options.velocityFilterWeight.value;
+        }
+
+        if (options.dotSize.changed) {
+            instance.sigpad.dotSize = options.dotSize.value;
+        }
+
+        if (options.readOnly.changed) {
+            if (options.readOnly.value === true) {
+                instance.sigpad.off();
+            }
+            else {
+                instance.sigpad.on();
+            }
+        }
+
+        if (options.imageType.changed) {
+            instance.options.imageType = options.imageType.value;
+        }
+
+        if (options.imageQuality.changed) {
+            instance.options.imageQuality = options.imageQuality.value;
+        }
+
+        if (options.includeImageBackgroundColor.changed) {
+            instance.options.includeImageBackgroundColor = options.includeImageBackgroundColor.value;
+        }
+    }
+}
+
+export function setData(element, elementId, data) {
+    const instance = _instances[elementId];
+
+    if (instance && instance.sigpad) {
+        instance.sigpad.fromData(data);
+    }
+}
+
+export function clear(element, elementId) {
+    const instance = _instances[elementId];
+
+    if (instance && instance.sigpad) {
+        instance.sigpad.clear();
+    }
+}
+
+function registerToEvents(dotNetAdapter, instance) {
+    if (instance && instance.sigpad) {
+        instance.sigpad.addEventListener("beginStroke", (e) => {
+            dotNetAdapter.invokeMethodAsync("NotifyBeginStroke")
+        });
+
+        instance.sigpad.addEventListener("endStroke", (e) => {
+            const dataURL = getImageDataURL(instance.sigpad, instance.options);
+
+            dotNetAdapter.invokeMethodAsync("NotifyEndStroke", dataURL)
+                .catch((reason) => {
+                    console.error(reason);
+                });
```

Comment:
Why is this swallowing an exception here??? Are we hiding a subtle bug? Also we should not console.error.

Reply:
removed